pokemongo_bot/human_behaviour.py

```diff
@@ -1,17 +1,175 @@
+    '''
+    lazy initialization of model
+    '''
+    if not hasattr(lognormal_model, 'delay_range2mu_sigma'):
+        lognormal_model.delay_range2mu_sigma = defaultdict(dict)
+        for line in open(path):
+            delay, delay_range, mu_, sigma = line.strip().split(',')
+            lognormal_model.delay_range2mu_sigma \
+                    [int(delay)][int(delay_range)] = \
+                    (float(mu_), float(sigma))
+    return lognormal_model.delay_range2mu_sigma
 
 
-def random_lat_long_delta():
-    # Return random value from [-.000025, .000025]. Since 364,000 feet is equivalent to one degree of latitude, this
-    # should be 364,000 * .000025 = 9.1. So it returns between [-9.1, 9.1]
-    return ((random() * 0.00001) - 0.000005) * 5
+def pareto(alpha=5.0, cap=10):
+    '''
+    somehow reflects human behaviour (Zipf's law)
+    transformed to return [alpha-1/alpha, inf) with mean of 1.0
+    capped to prevent inf
+    '''
+    alpha = float(alpha)
+    return min((alpha - 1) / alpha / ((1 - random())**(1 / alpha)), cap)
+
+
+def lognormal(mu_, sigma):
+    '''
+    Good assumption for ping, human reflexes.
+    mu and sigma are for the lognormal, not the values you see.
+    Use model parameter to get what you want.
+    '''
+    return exp(gauss(mu_, sigma))
+
+
+def sleep(seconds):
+    '''
+    sleep for (given seconds + jitter + human reflex) seconds.
+    '''
+    time.sleep(seconds + jitter() + human_reflex())
+
+
+def jitter(ping=80, ping_range=30):
+    '''
+    Simple lognormal jitter model for given ping and range.
+    Actually wider range is preferred for accurate modeling,
+    but we don't want to spend much time...
+    '''
+    ping = int(ping / 10) * 10
+    ping_range = int(ping_range / 5) * 5
+    ping = min(max(10, ping), 500)
+    ping_range = min(max(10, ping_range), 100)
+    mu_, sigma = lognormal_model()[ping][ping_range]
+    lag = lognormal(mu_, sigma)
+    lag = min(1.0, lag)
+    return lag
+
+
+def human_reflex(reflex_mean=270, reflex_range=100):
+    '''
+    Simulates human reflexes.
+    '''
+    mu_, sigma = lognormal_model()[reflex_mean][reflex_range]
+    return min(1.0, lognormal(mu_, sigma))
+
+
+def action_delay(low, high):
+    # Waits for random number of seconds between low & high numbers
+    longNum = uniform(low, high)
+    shortNum = float("{0:.2f}".format(longNum))
+    time.sleep(shortNum)
+
+
+def random_lat_long_delta(radius=0.00025):
+    '''
+    Simulates gps error.
+    '''
+    # Return random value from [-.000025, .000025]ish 99.73% of time.
+    # Since 364,000 feet is equivalent to one degree of latitude, this
+    # Gaussian is better for gps errors
+    error = gauss(0, radius/3.0)
+    error = min(max(-radius, error), radius)
+    return error
+
+
+# Humanized `normalized_reticle_size` parameter for `catch_pokemon` API.
+# 1.0 => normal, 1.950 => excellent
+def normalized_reticle_size(factor):
+    minimum = 1.0
+    maximum = 1.950
+    return uniform(
```

Comment:
Gps errors should be approximated by gaussian.

I'm not quite sure for reticle and spin rngs. 
Speaking of myself, I try to get simple 'Great' and don't throw spin balls because I'm bad at it.
However, that might be my preference and most of the users might be acting otherwise.
Besides, I'm also not certain what distributions are proper at this point.

It should be other than uniform to prevent us from getting banned in the future, but I decided to keep it as it is for now, before going deeper and messing around with the codes.